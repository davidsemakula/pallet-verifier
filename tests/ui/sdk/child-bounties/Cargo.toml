[package]
name = "pallet-child-bounties"
version = "27.0.0"
#authors.workspace = true
#edition.workspace = true
edition = "2021"
license = "Apache-2.0"
#homepage.workspace = true
#repository.workspace = true
description = "FRAME pallet to manage child bounties"
readme = "README.md"

#[lints]
#workspace = true

[package.metadata.docs.rs]
targets = ["x86_64-unknown-linux-gnu"]

[dependencies]
codec = { version = "3.6.12", package = "parity-scale-codec", default-features = false, features = ["derive"] }
log = { version = "0.4.22", default-features = false }
scale-info = { version = "2.11.3", default-features = false, features = ["derive"] }
frame-benchmarking = { version = "38.0.0", optional = true, default-features = false }
frame-support = { version = "38.0.0", default-features = false }
frame-system = { version = "38.0.0", default-features = false }
pallet-bounties = { version = "37.0.0", default-features = false }
pallet-treasury = { version = "37.0.0", default-features = false }
sp-core = { version = "34.0.0", default-features = false }
sp-io = { version = "38.0.0", default-features = false }
sp-runtime = { version = "39.0.0", default-features = false }

[dev-dependencies]
pallet-balances = { version = "39.0.0", default-features = true }

[features]
default = ["std"]
std = [
	"codec/std",
	"frame-benchmarking?/std",
	"frame-support/std",
	"frame-system/std",
	"log/std",
	"pallet-balances/std",
	"pallet-bounties/std",
	"pallet-treasury/std",
	"scale-info/std",
	"sp-core/std",
	"sp-io/std",
	"sp-runtime/std",
]
runtime-benchmarks = [
	"frame-benchmarking",
	"frame-benchmarking/runtime-benchmarks",
	"frame-support/runtime-benchmarks",
	"frame-system/runtime-benchmarks",
	"pallet-balances/runtime-benchmarks",
	"pallet-bounties/runtime-benchmarks",
	"pallet-treasury/runtime-benchmarks",
	"sp-runtime/runtime-benchmarks",
]
try-runtime = [
	"frame-support/try-runtime",
	"frame-system/try-runtime",
	"pallet-balances/try-runtime",
	"pallet-bounties/try-runtime",
	"pallet-treasury/try-runtime",
	"sp-runtime/try-runtime",
]
