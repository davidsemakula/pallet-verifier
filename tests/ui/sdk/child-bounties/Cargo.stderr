warning: possible attempt to add with overflow
  --> src/lib.rs:587:4
   |
LL | /             ChildBounties::<T>::try_mutate_exists(
LL | |                 parent_bounty_id,
LL | |                 child_bounty_id,
LL | |                 |maybe_child_bounty| -> DispatchResult {
...  |
LL | |                 },
LL | |             )?;
   | |_____________^
   |
note: related location
  --> /Users/david/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/frame-support-38.2.0/src/storage/types/double_map.rs:415:3
   |
LL |         <Self as crate::storage::StorageDoubleMap<Key1, Key2, Value>>::try_mutate_exists(k1, k2, f)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: related location
  --> /Users/david/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/frame-support-38.2.0/src/storage/generator/double_map.rs:302:13
   |
LL |         let ret = f(&mut val);
   |                   ^^^^^^^^^^^
note: related location
  --> src/lib.rs:604:19
   |
LL |   ...                   unlock_at: frame_system::Pallet::<T>::block_number() +
   |  __________________________________^
LL | | ...                       T::BountyDepositPayoutDelay::get(),
   | |____________________________________________________________^
note: related location
  --> /Users/david/.rustup/toolchains/nightly-2025-01-10-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/arith.rs:111:1
   |
LL | add_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f16 f32 f64 f128 }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this warning originates in the macro `add_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

